components/url/origin.rs

```diff
@@ -101,7 +101,7 @@ pub struct OpaqueOrigin(Uuid);
 known_heap_size!(0, OpaqueOrigin);
 
 /// A representation of an [origin](https://html.spec.whatwg.org/multipage/#origin-2).
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
```

Comment:
So when using the new `origin` method on `GlobalScope`:
```
    /// Get the origin for this global scope
    pub fn origin(&self) -> &MutableOrigin {
        &self.origin
    }
```
I'll be doing something like this:
```
origin: GlobalScope::origin().immutable().clone()
```
Is that the idea?